genext2fs.c

```diff
@@ -2907,7 +2907,7 @@ add2fs_from_dir(filesystem *fs, uint32 this_nod, int squash_uids, int squash_per
 			}
 		}
 	}
-	closedir(dh);
+	free(dents);
```

Comment:
This leaks memory. You need to free `dents[i]` as well. See the example at the bottom of the `scandir(3)` manual page.